File: songs/go/06-flatten.go
package main

import (
	"example.com/mysongs"
	"fmt"
)

func flatten(songs []mysongs.Song) []string {
	var tags []string

	for _, song := range songs {
		if song.Tags != nil {
			for _, tag := range song.Tags {
				tags = append(tags, tag)
			}
		}
	}

	return tags
}

func main() {
	tags := flatten(mysongs.GetSongs())
	fmt.Println(tags)
}

File: songs/rust/06-flatten.rs
mod my_songs;

fn main() {
    let songs: Vec<my_songs::Song> =
        my_songs::get_songs();

    let tags: Vec<&str> = songs
        .into_iter()
        .filter_map(|song| song.tags)
        .flat_map(|tag_s| tag_s)
        .collect();
    
    println!("{:?}", tags);
}
